.github/workflows/fitness-functions.yml

```
@@ -24,6 +26,9 @@ jobs:
         env:
           BASE_REF: ${{ github.event.pull_request.base.ref }}
         run: |
-          git fetch origin $BASE_REF
-          git diff origin/$BASE_REF HEAD -- . > diff
+          # The following command generates a diff of changes between the common
+          # ancestor of $BASE_REF and HEAD, and the current commit (HEAD), for
+          # files in the current directory and its subdirectories. The output is
+          # then saved to a file called "diff".
+          git diff "$(git merge-base origin/$BASE_REF HEAD)" HEAD -- . > diff
```

I just tried running this through [shellcheck](https://www.shellcheck.net/), and it also suggested two more changes:

```suggestion
          git diff "$(git merge-base "origin/$BASE_REF" HEAD)" HEAD -- . > ./diff
```

The quoting helps ensure this works correctly if `$BASE_REF` includes spaces somehow, and the `./` in front of `diff` ensures that this content is sent to the _file_ `diff` rather than the command `diff`.